File: javascript/fp-patterns/immediately-invoked-function-expression/IIFE-with-return-value/IIFE-with-return-value.js
// IIFE с возвращаемым значением
const singleton = ((str) => {
  console.log(`${str} от IIFE, создающей одиночку`);
  console.log('Создаю одиночку');
  return {
    name: 'Одиночка',
  };
})('Привет');

console.log('Меня зовут', singleton.name);

File: go/fp-patterns/immediately-invoked-function-expression/IIFE-with-return-value/IIFE-with-return-value.go
package main

import "fmt"

// Singleton - гипотетический объект-одиночка
type Singleton struct{ name string }

// IIFE с возвращаемым значением
func main() {
	singleton := func(str string) Singleton {
		fmt.Println(str, "от IIFE, создающей одиночку")
		fmt.Println("Создаю одиночку")
		return Singleton{
			name: "Одиночка",
		}
	}("Привет")
	fmt.Println("Меня зовут", singleton.name)
}
